extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/base/ModeledDirectoryObjectService.java

```diff
@@ -387,26 +398,38 @@ public Collection<InternalType> retrieveObjects(ModeledAuthenticatedUser user,
             Collection<String> identifiers) throws GuacamoleException {
 
         // Ignore invalid identifiers
-        identifiers = filterIdentifiers(identifiers);
+        List<String> filteredIdentifiers = filterIdentifiers(identifiers);
 
         // Do not query if no identifiers given
-        if (identifiers.isEmpty())
+        if (filteredIdentifiers.isEmpty())
             return Collections.<InternalType>emptyList();
 
-        Collection<ModelType> objects;
+        int batchSize = environment.getDefaultBatchSize();
 
-        // Bypass permission checks if the user is privileged
-        if (user.isPrivileged())
-            objects = getObjectMapper().select(identifiers);
+        boolean userIsPrivileged = user.isPrivileged();
+
+        // Process the filteredIdentifiers in batches using IntStream.range and flatMap
+        Collection<ModelType> allObjects = IntStream.range(0, filteredIdentifiers.size())
+                .filter(index -> index % batchSize == 0)
```

Comment:
While I do like Java's fancy streams, this seems an inefficient way to generate the starting indices of batches. I think we should directly calculate the starting indices rather than doing a brute-force search through all indices.

Reply:
@mike-jumper see also the conversation regarding the use of https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Lists.html#partition(java.util.List,%20int).

Reply:
Updated to use `Lists.partition`